### ISSUE TITLE
OCS4 didn't scale

### ISSUE THREAD

**Issue by christianh814** [openshift/openshift-cns-testdrive#439]
I wasn't able to get another OCS4 node by "just scaling the OCS node" I still only ended up with 3 ceph instances

```
sh-4.2# ceph osd status
+----+--------------------------------------------+-------+-------+--------+---------+--------+---------+-----------+
| id |                    host                    |  used | avail | wr ops | wr data | rd ops | rd data |   state   |
+----+--------------------------------------------+-------+-------+--------+---------+--------+---------+-----------+
| 0  | ip-10-0-136-13.us-east-2.compute.internal  | 1067M | 68.8G |    0   |     0   |    0   |     0   | exists,up |
| 1  | ip-10-0-169-14.us-east-2.compute.internal  | 1054M | 68.8G |    0   |     0   |    0   |     0   | exists,up |
| 2  | ip-10-0-154-201.us-east-2.compute.internal | 1071M | 68.8G |    0   |     0   |    0   |     0   | exists,up |
+----+--------------------------------------------+-------+-------+--------+---------+--------+---------+-----------+
sh-4.2# ceph status
  cluster:
    id:     69b85eea-0956-423f-a0d5-07cde3908c02
    health: HEALTH_OK

  services:
    mon: 3 daemons, quorum c,b,d (age 16h)
    mgr: a(active, since 16h)
    osd: 3 osds: 3 up, 3 in

  data:
    pools:   1 pools, 100 pgs
    objects: 36 objects, 73 MiB
    usage:   3.1 GiB used, 206 GiB / 210 GiB avail
    pgs:     100 active+clean
```

I scaled the machinesets as described in the doc

```
[~] $ oc get machines -n openshift-machine-api -l machine.openshift.io/cluster-api-machine-role=workerocs
NAME                                            INSTANCE              STATE     TYPE        REGION      ZONE         AGE
cluster-4b71-l7bvd-workerocs-us-east-2a-4qqvt   i-0db45fa2644abbef6   running   m5d.large   us-east-2   us-east-2a   16h
cluster-4b71-l7bvd-workerocs-us-east-2a-gndfn   i-0765abbf1d6180ba9   running   m5d.large   us-east-2   us-east-2a   17h
cluster-4b71-l7bvd-workerocs-us-east-2b-npsfw   i-0b17e291bd24cff0b   running   m5d.large   us-east-2   us-east-2b   17h
cluster-4b71-l7bvd-workerocs-us-east-2c-6kzqw   i-0391935f2efb1c3d6   running   m5d.large   us-east-2   us-east-2c   17h
```
Not sure if this is one of those "60% of the time it works every time" thing, or if it's something that SHOULD work but didn't in my particular care (for whatever reason).

If it SHOULD work, then we should write some notes about "how to debug"

**Comment by christianh814** [comment]
CC: @kaovilai

Emailing OCS team; but CC-ing you for visibility

**Comment by netzzer** [comment]
See https://github.com/rook/rook/issues/2977. For now the rook operator pod needs to restarted (deleted) to get the osd pod created on the new OCP node. This should be fixed in OCS 4.2 w/OCP 4.2.

**Comment by kaovilai** [comment]
@thoraxe Do you want the workaround `rook operator pod needs to restarted` to be added to dev?
Will 4.2 come out before Tech Exchange?

> See [rook/rook#2977](https://github.com/rook/rook/issues/2977). For now the rook operator pod needs to restarted (deleted) to get the osd pod created on the new OCP node. This should be fixed in OCS 4.2 w/OCP 4.2.

**Comment by christianh814** [comment]
> @thoraxe Do you want the workaround `rook operator pod needs to restarted` to be added to dev?
> Will 4.2 come out before Tech Exchange?
> 
> > See [rook/rook#2977](https://github.com/rook/rook/issues/2977). For now the rook operator pod needs to restarted (deleted) to get the osd pod created on the new OCP node. This should be fixed in OCS 4.2 w/OCP 4.2.

FWIW @kaovilai ...I did test it (by deleting the rook operator pod) and it does work after doing that.

```
sh-4.2# ceph status
  cluster:
    id:     69b85eea-0956-423f-a0d5-07cde3908c02
    health: HEALTH_OK
 
  services:
    mon: 3 daemons, quorum c,b,d (age 34h)
    mgr: a(active, since 8m)
    osd: 4 osds: 4 up, 4 in
 
  data:
    pools:   1 pools, 100 pgs
    objects: 36 objects, 73 MiB
    usage:   4.1 GiB used, 274 GiB / 279 GiB avail
    pgs:     100 active+clean
 
  io:
    client:   1.3 KiB/s wr, 0 op/s rd, 0 op/s wr
 
sh-4.2# ceph osd status
+----+--------------------------------------------+-------+-------+--------+---------+--------+---------+-----------+
| id |                    host                    |  used | avail | wr ops | wr data | rd ops | rd data |   state   |
+----+--------------------------------------------+-------+-------+--------+---------+--------+---------+-----------+
| 0  | ip-10-0-136-13.us-east-2.compute.internal  | 1059M | 68.8G |    0   |     0   |    0   |     0   | exists,up |
| 1  | ip-10-0-169-14.us-east-2.compute.internal  | 1050M | 68.8G |    0   |     0   |    0   |     0   | exists,up |
| 2  | ip-10-0-154-201.us-east-2.compute.internal | 1066M | 68.8G |    0   |     0   |    0   |     0   | exists,up |
| 3  | ip-10-0-130-108.us-east-2.compute.internal | 1043M | 67.9G |    0   |     0   |    0   |     0   | exists,up |
+----+--------------------------------------------+-------+-------+--------+---------+--------+---------+-----------+
```

I think it's worth it to make a note about it in the doc since the first two TEs will be before the release of OCS/OCP 4.2

**Comment by kaovilai** [comment]
Will be adding that instruction!
